### PR TITLE
[Portafly] Updating PF4 dependencies and fixing tests

### PR DESCRIPTION
**Pull request by damianpm**
**What this PR does / why we need it**:

Patternfly/React 4.12.7 is released with some [breaking changes](https://github.com/redallen/patternfly-scripts/blob/master/github/patternfly/patternfly-react/pull_requests/breaking-change-notes.md), we need to update dependencies, some components and tests

**Which issue(s) this PR fixes** 
None

**Verification steps** 
Go to portafly dir:
```
cd portafly
```
Install dependencies:
```
yarn
```

Start in browser
```
yarn start
```
Run tests:
```
yarn test
```

Notes:
@cntlsn @thomasmaas
This new release introduces some visual changes, please take a look

### CODE REVIEW

File: portafly/src/components/AppLayout.tsx [hunk 1]
```diff
@@ -2,7 +2,6 @@ import * as React from 'react'
 import {
   Nav,
   NavList,
-  NavVariants,
```
Comment (damianpm) [inline portafly/src/components/AppLayout.tsx:5]: `NavVariants` was deleted in https://github.com/patternfly/patternfly-react/pull/4225

File: portafly/src/tests/components/AppLayout.test.tsx [hunk 2]
```diff
@@ -56,18 +56,19 @@ describe('AppLayout tests', () => {
     expect(getByLabelText('Global navigation')).toBeInTheDocument()
   })
 
-  it('should hide the sidebar when clicking the nav-toggle button', async () => {
-    const { getByLabelText, getByTestId } = renderAppLayout()
-    const navButton = getByLabelText('Global navigation')
+  it('should start with collapsed sidebar', async () => {
```
Comment (damianpm) [inline portafly/src/tests/components/AppLayout.test.tsx:59]: It should start collapsed by default, because of mobile-first approach defined [here](https://github.com/3scale/porta/blob/master/portafly/src/components/AppLayout.tsx#L51)

File: portafly/src/tests/components/AppLayout.test.tsx [hunk 3]
```diff
@@ -56,18 +56,19 @@ describe('AppLayout tests', () => {
...
     expect(sidebar).toHaveClass('pf-m-collapsed')
   })
 
-  it('should start with an hidden sidebar', async () => {
-    const { getByTestId } = renderAppLayout({ startWithOpenNav: false })
+  it('should show the sidebar when clicking the nav-toggle button', async () => {
+    const { getByLabelText, getByTestId } = renderAppLayout()
+    const navButton = getByLabelText('Global navigation')
     const sidebar = getByTestId('app-sidebar')
     expect(sidebar).toHaveClass('pf-m-collapsed')
```
Comment (josemigallas) [inline portafly/src/tests/components/AppLayout.test.tsx:69]: Since this is being asserted twice (here and in the previous case) we could leave only the second, renaming it to something like `'should be hidden by default and be able to open'`.

**Comment by didierofrivia** [comment]
Change the base to `portafly/milestone_dev-accounts` and run the tests,  this should be pointed to the milestone branch instead of `master`

File: portafly/src/tests/components/AppLayout.test.tsx (hunk 3 above)
Comment (damianpm) [reply portafly/src/tests/components/AppLayout.test.tsx:63]: done!

File: portafly/src/components/data-list/SearchWidget.tsx [hunk 4]
```diff
@@ -196,10 +194,10 @@ const SearchWidget: React.FunctionComponent<ISearchWidget> = ({ categories }) =>
...
 
   return (
-    <DataToolbarGroup variant="filter-group">
-      { categorySelect }
-      { appliedFilters }
-      { selectedCategory.options
+    <ToolbarGroup variant="filter-group">
+      {categorySelect}
+      {appliedFilters}
+      {selectedCategory.options
```
Comment (didierofrivia) [inline portafly/src/components/data-list/SearchWidget.tsx:200]: These unnecessary changes were triggered by tslint? if not, avoid them

File: portafly/src/components/util/AlertsContext.tsx [hunk 5]
```diff
@@ -14,7 +14,7 @@ interface IAlert {
 }
 
 type IAlertsContext = { addAlert: (alert: IAlert) => void }
-const AlertsContext = React.createContext<IAlertsContext>({ addAlert: () => {} })
+const AlertsContext = React.createContext<IAlertsContext>({ addAlert: () => { } })
```
Comment (didierofrivia) [inline portafly/src/components/util/AlertsContext.tsx:17]: Same here, and the rest of the PR. Even if it was auto-triggered, it should not pollute the reason of this PR

File: portafly/src/components/data-list/Toolbar.tsx [hunk 6]
```diff
@@ -1,5 +1,5 @@
 import React from 'react'
-import { DataToolbar } from '@patternfly/react-core/dist/js/experimental'
+import { Toolbar as DataToolbar } from '@patternfly/react-core'
```
Comment (didierofrivia) [inline portafly/src/components/data-list/Toolbar.tsx:2]: Right, our naming is colliding with PF... when we are done with the DataList, we might rename this Toolbar to something else

Comment (josemigallas) [reply portafly/src/components/data-list/Toolbar.tsx:2]: Wait.. we would want to take a look at this. We were using `DataToolbar` only, now they've merged `Toolbar` and `DataToolbar`. Maybe the markup/usage has changed too. Docs are not updated 😂

File: portafly/src/components/util/AlertsContext.tsx (hunk 5 above)
Comment (josemigallas) [reply portafly/src/components/util/AlertsContext.tsx:17]: This is not be triggered by our eslint ruleset anyway.

File: portafly/src/components/data-list/Toolbar.tsx (hunk 6 above)
Comment (josemigallas) [reply portafly/src/components/data-list/Toolbar.tsx:2]: I think it would be better to rename our own to `DataListToolbar` then, maybe export it as `Toolbar` to keep changes to a minimum.

File: portafly/src/components/data-list/SearchWidget.tsx (hunk 4 above)
Comment (damianpm) [reply portafly/src/components/data-list/SearchWidget.tsx]: done!

File: portafly/src/components/util/AlertsContext.tsx (hunk 5 above)
Comment (damianpm) [reply portafly/src/components/util/AlertsContext.tsx]: done

File: portafly/src/components/data-list/Toolbar.tsx (hunk 6 above)
Comment (damianpm) [reply portafly/src/components/data-list/Toolbar.tsx:2]: PF is not using `DataToolbar` anymore, all the components that started with `Data...` are renamed: `DataToolbarGroup` -> `ToolbarGroup`, etc.

File: portafly/src/components/data-list/SearchWidget.tsx [hunk 7]
```diff
@@ -53,11 +51,11 @@ const CategorySelect: React.FunctionComponent<ICategorySelect> = ({
...
       onSelect={onSelect}
       selections={selections}
-      isExpanded={isExpanded}
-      onToggle={setIsExpanded}
+      isOpen={isOpen}
+      onToggle={setIsOpen}
       isDisabled={false}
     >
-      { categories.map((category) => (
+      {categories.map((category) => (
```
Comment (didierofrivia) [inline portafly/src/components/data-list/SearchWidget.tsx:58]: 👎

File: portafly/src/components/data-list/SearchWidget.tsx [hunk 8]
```diff
@@ -90,13 +88,13 @@ const OptionsSelect: React.FunctionComponent<ICollectionSelect> = ({
...
       onSelect={onSelect}
       selections={selections}
-      isExpanded={isExpanded}
-      onToggle={setIsExpanded}
+      isOpen={isOpen}
+      onToggle={setIsOpen}
       placeholderText={`Filter by ${categoryName}`}
     >
-      { options.map(({ name, humanName }) => (
+      {options.map(({ name, humanName }) => (
```
Comment (didierofrivia) [inline portafly/src/components/data-list/SearchWidget.tsx:95]: 👎

File: portafly/src/components/data-list/SearchWidget.tsx [hunk 9]
```diff
@@ -90,13 +88,13 @@ const OptionsSelect: React.FunctionComponent<ICollectionSelect> = ({
...
         <SelectOption key={name} value={humanName} />
-      )) }
+      ))}
```
Comment (didierofrivia) [inline portafly/src/components/data-list/SearchWidget.tsx:97]: 👎

File: portafly/src/components/data-list/Toolbar.tsx (hunk 6 above)
Comment (damianpm) [reply portafly/src/components/data-list/Toolbar.tsx]: Renamed!
